tests/testthat/test-TealData.R

```diff
@@ -1,238 +1,342 @@
+  jk <- join_keys(join_key("df1", "df2", "id"))
+  data <- TealData$new(df1, df2, join_keys = jk, check = FALSE)
 
-    x <- teal.data:::TealData$new(x1, x2)
-  })
+  testthat::expect_equal(
+    data$get_join_keys(),
+    join_keys(
+      join_key("df1", "df2", "id"),
+      join_key("df2", "df1", "id")
+    )
+  )
+})
 
-  expect_silent({
-    x_copy <- x$clone(deep = TRUE)
-  })
+testthat::test_that("TealData$get_join_keys returns all join_keys of the single dataname specified", {
+  df1 <- data.frame(id = c("A", "B"), a = c(1L, 2L))
+  df2 <- data.frame(df2_id = c("A", "B"), id = c("A", "B"), b = c(1L, 2L))
 
-  expect_silent({
-    lapply(x$get_items(), load_dataset)
-  })
+  df1 <- dataset("df1", df1, keys = "id")
+  df2 <- dataset("df2", df2, keys = "df2_id")
 
-  expect_true(is_pulled(x))
-  expect_false(is_pulled(x_copy))
+  jk <- join_keys(join_key("df1", "df2", "id"))
+  data <- TealData$new(df1, df2, join_keys = jk, check = FALSE)
 
-  # check one of the private fields of list of R6 class object
-  expect_false(rlang::is_reference(x$get_items()[[1]], x_copy$get_items()[[1]]))
-  # check one of the private fields of R6 class
-  expect_false(rlang::is_reference(x$get_join_keys(), x_copy$get_join_keys()))
+  testthat::expect_equal(
+    data$get_join_keys("df1"),
+    list(df2 = setNames("id", "id"))
+  )
 })
 
-testthat::test_that("execute_mutate returns current datasets if no mutate_code", {
-  pull_fun <- callable_function(data.frame)
-  pull_fun$set_args(args = list(head_letters = head(letters)))
-  t_dc <- dataset_connector("test_dc", pull_fun)
-  t_ds <- dataset("head_rock", head(rock), code = "head_rock <- head(rock)") %>%
-    mutate_dataset("head_rock$head_letters <- test_dc$head_letters", vars = list(test_dc = t_dc))
-  data <- teal_data(t_dc, t_ds)
-  testthat::expect_identical(
-    data$execute_mutate(),
-    list(head_rock = t_ds)
+testthat::test_that("TealData$get_join_keys returns the join_keys of the specified datanames", {
+  df1 <- data.frame(id = c("A", "B"), a = c(1L, 2L))
+  df2 <- data.frame(df2_id = c("A", "B"), id = c("A", "B"), b = c(1L, 2L))
+
+  df1 <- dataset("df1", df1, keys = "id")
+  df2 <- dataset("df2", df2, keys = "df2_id")
+
+  jk <- join_keys(join_key("df1", "df2", "id"))
+  data <- TealData$new(df1, df2, join_keys = jk, check = FALSE)
+
+  testthat::expect_equal(
+    data$get_join_keys("df1", "df2"),
+    setNames("id", "id")
   )
 })
 
-# Multiple connectors ----
-testthat::test_that("Multiple connectors wrapped in cdisc_data", {
-  example_data_connector <- function(...) {
-    connectors <- list(...)
-    open_fun <- callable_function(library)
-    open_fun$set_args(list(package = "teal"))
-    con <- TealDataConnection$new(open_fun = open_fun)
-    TealDataConnector$new(connection = con, connectors = connectors)
-  }
+testthat::test_that("TealData$get_parents returns an empty list even when parents are specified", {
+  # parent specified - this information is passed on the wrappers level
```

Comment:
I think comment should say:

Parent information is passed through the join_keys argument - this test does not create creating TealData with such a join_keys object set (it is only set for you if you call teal/cdisc_data)

Reply:
done.